### ISSUE TITLE
Dashboard plots fail: Value must be an integer

### ISSUE THREAD

**Issue by kyleabeauchamp** [metabase/metabase#9640]
- Chrome 73.0
- OSX 10.14
- postgres 11.2
- Metabase 32.0
- Metabase docker
- Metabase internal database H2

On the summary dashboard for very large tables (>25M rows), I'm seeing "There was a problem displaying this chart" for nearly every plot that ought to display.  On dashboards for smaller tables (when DB load is low), plots display normally.  I looked at the server logs, and the only clue I can find is this:

```
Mar 29 19:02:14 DEBUG metabase.middleware.log :: POST /api/dataset 400 (1 ms) (0 DB calls).
{:errors {:database "value must be an integer."}}
```
My theory is that the DB query is timing out but not bubbling the exception up in a transparent manner.

**Comment by kyleabeauchamp** [comment]
When I query the same table but from the "Question" interface, I get a "Your question took too long" error, which is what I expect.

**Comment by camsaul** [comment]
That error suggests that the query it's sending in doesn't contain the ID of the database to run the query against, which is strange -- it should never have saved that query in the first place. Can you try creating a new question with the same Query and see if that works?

**Comment by kyleabeauchamp** [comment]
I tried to replicate an identical query using the question interface and I get "Your question took too long", but it also shows the same `value must be an integer.` error in the logs.  This occurs both with and without caching enabled.

**Comment by initd3v** [comment]
I can confirm this error. I've created a native query on 0.31 to a Postgres database - it worked fine for me. After upgrading to 0.32.1 the question is still working. But if I change a condition (a date) in the where clause it is running for a short while and keeps saying "Your question took too long". In the logs occur an error:

Apr 02 08:58:04 DEBUG metabase.middleware.log :: POST /api/dataset 400 (1 ms) (0 DB calls).
{:errors {:database "value must be an integer."}}

I've tried to copy that changed query to a new question and also tried to give some JDBC parameters to the database connection for set up a higher connection timeout. Nothing worked.

Maybe there exist an error with the async requests?

**Comment by nborrmann** [comment]
I am observing the same issue. `/api/dataset` queries time out after 30 seconds and return `{"errors":{"database":"value must be an integer."}}` as a response.

However, on saved queries (`/api/card/{id}/query` endpoint) there is no timeout, and queries happily return data after >30s. 

So this cannot be a database or network timeout.

**Comment by Zatvobor** [comment]
run 'New Question' and got "Your question took too long"
```
SELECT COUNT(*) FROM responses WHERE organization_id=373;
```
```
04-09 15:59:22 DEBUG middleware.log :: POST /api/dataset 400 (1 ms) (0 DB calls).
{:errors {:database "value must be an integer."}}
```

**Comment by kyleabeauchamp** [comment]
This bug and the hardcoded timeout is a major blocker for my own usage of metabase.

**Comment by kyleabeauchamp** [comment]
It's possible this is fixed by https://github.com/metabase/metabase/pull/9704 and available as part of 0.32.3.  I think it's resolved some of my issues, but I'll leave this open as others confirm the fix.

**Comment by initd3v** [comment]
I can confirm that the current version 0.32.4 resolves the issue - further more the csv export is working again and does not present the csv data as raw output on the website.
